identity_stardust/src/client/client_ext.rs

```diff
@@ -0,0 +1,616 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::borrow::Cow;
+use std::ops::Deref;
+
+use identity_did::did::DIDError;
+use iota_client::api_types::responses::OutputResponse;
+use iota_client::block::address::Address;
+use iota_client::block::output::feature::IssuerFeature;
+use iota_client::block::output::feature::SenderFeature;
+use iota_client::block::output::unlock_condition::AddressUnlockCondition;
+use iota_client::block::output::unlock_condition::GovernorAddressUnlockCondition;
+use iota_client::block::output::unlock_condition::StateControllerAddressUnlockCondition;
+use iota_client::block::output::AliasId;
+use iota_client::block::output::AliasOutput;
+use iota_client::block::output::AliasOutputBuilder;
+use iota_client::block::output::BasicOutputBuilder;
+use iota_client::block::output::Feature;
+use iota_client::block::output::Output;
+use iota_client::block::output::OutputId;
+use iota_client::block::output::RentStructure;
+use iota_client::block::output::UnlockCondition;
+use iota_client::block::payload::transaction::TransactionEssence;
+use iota_client::block::payload::Payload;
+use iota_client::block::Block;
+use iota_client::secret::SecretManager;
+use iota_client::Client;
+
+use crate::error::Result;
+use crate::Error;
+use crate::NetworkName;
+use crate::StardustDID;
+use crate::StardustDocument;
+
+/// An extension trait for a [`Client`] that provides helper functions for publication
+/// and resolution of DID documents in Alias Outputs.
+///
+/// This trait is only meant to be used rather than implemented.
+#[async_trait::async_trait]
+pub trait StardustClientExt: Sync {
+  /// Returns a reference to a [`Client`].
+  fn client(&self) -> &Client;
+
+  /// Create a DID with a new Alias Output containing the given `document`.
+  ///
+  /// The `address` will be set as the state controller and governor unlock conditions.
+  /// The minimum required token deposit amount will be set according to the given
+  /// `rent_structure`, which will be fetched from the node if not provided.
+  /// The returned Alias Output can be further customized before publication, if desired.
+  ///
+  /// NOTE: this does *not* publish the Alias Output. See [`publish_did_output`](StardustClientExt::publish_did_output).
+  ///
+  /// # Errors
+  ///
+  /// - Returns an [`Error::DIDUpdateError`] when retrieving the `RentStructure` fails.
+  /// - Returns an [`Error::AliasOutputBuildError`] when building the Alias Output fails.
+  async fn new_did(
```

Comment:
Both `new_did` and `update_did` return `AliasOutput`, should they be named `*_did_output` to be consistent with the other methods?